modules/public/logic.py

```diff
@@ -0,0 +1,258 @@
+import bleach
+from math import floor
+import pycountry
+import re
+from urllib.parse import parse_qs, urlparse
+
+
+def calculate_colors(countries, max_users):
+    """Calculate the displayed colors for a list of country depending on the
+    maximum of users
+
+    :param countries: List of countries
+    :param max_users: Maximum of users
+
+    :returns: The list of countries with a caculated color for each
+    """
+    for country_code in countries:
+        countries[country_code]['color_rgb'] = [
```

Comment:
It's a bad pattern to directly manipulate a variable that was passed by reference - functions should [be atomic and avoid side-effects](https://softwareengineering.stackexchange.com/questions/15269/why-are-side-effects-considered-evil-in-functional-programming) to avoid confusion.

So what you should do in this case is create a new dictionary and return that instead.